pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -86,11 +86,11 @@ func SetDefaults_SeedNetworks(obj *SeedNetworks) {
 
 // SetDefaults_SeedSettingDependencyWatchdog sets defaults for SeedSettingDependencyWatchdog objects.
 func SetDefaults_SeedSettingDependencyWatchdog(obj *SeedSettingDependencyWatchdog) {
-	if obj.Endpoint == nil {
-		obj.Endpoint = &SeedSettingDependencyWatchdogEndpoint{Enabled: true}
+	if obj.Endpoint == nil && obj.Weeder == nil {
```

Comment:
Why do you couple this? It's better to keep it in separate conditions (otherwise, `Weeder` might not be defaulted when `endpoint` is non-`nil`). same below and for `v1beta1`